src/kfactory/kcell.py

```diff
@@ -1484,6 +1484,317 @@ class Constants(BaseSettings):
     pass
 
 
+class LayerLevel(BaseModel):
+    """Level for 3D LayerStack.
+
+    Parameters:
+        layer: (GDSII Layer number, GDSII datatype).
+        thickness: layer thickness in dbu.
+        thickness_tolerance: layer thickness tolerance in dbu.
+        zmin: height position where material starts in dbu.
+        material: material name.
+        sidewall_angle: in degrees with respect to normal.
+        z_to_bias: parametrizes shrinking/expansion of the design GDS layer
+            when extruding from zmin (0) to zmin + thickness (1).
+            Defaults no buffering [[0, 1], [0, 0]].
+        info: simulation_info and other types of metadata.
+            mesh_order: lower mesh order (1) will have priority over higher
+                mesh order (2) in the regions where materials overlap.
+            refractive_index: refractive_index
+                can be int, complex or function that depends on wavelength (um).
+            type: grow, etch, implant, or background.
+            mode: octagon, taper, round.
+                https://gdsfactory.github.io/klayout_pyxs/DocGrow.html
+            into: etch into another layer.
+                https://gdsfactory.github.io/klayout_pyxs/DocGrow.html
+            doping_concentration: for implants.
+            resistivity: for metals.
+            bias: in um for the etch.
+    """
+
+    layer: tuple[int, int] | LayerEnum
+    thickness: int
+    thickness_tolerance: int | None = None
+    zmin: int
+    material: str | None = None
+    sidewall_angle: int = 0
+    z_to_bias: tuple[int, ...] | None = None
+    info: Info = Info()
+
+
+class LayerStack(BaseModel):
+    """For simulation and 3D rendering.
+
+    Parameters:
+        layers: dict of layer_levels.
+    """
+
+    layers: dict[str, LayerLevel] = Field(default_factory=dict)
+
+    def __init__(self, **data: Any):
+        """Add LayerLevels automatically for subclassed LayerStacks."""
+        super().__init__(**data)
+
+        for field, val in data.items():
+            if not isinstance(val, LayerLevel):
+                raise TypeError(
+                    f"argument should be of type LayerLevel, got type {type(val)}."
+                )
+            self.layers[field] = val
+            if isinstance(val.layer, LayerEnum):
+                self.layers[field].layer = (val.layer[0], val.layer[1])
+
+    def get_layer_to_thickness(self) -> dict[tuple[int, int] | LayerEnum, int]:
+        """Returns layer tuple to thickness (um)."""
+        return {
+            level.layer: level.thickness
+            for level in self.layers.values()
+            if level.thickness
+        }
```

Comment:
Wouldn't it make more sense to have a layer thickness in µm instead of dbu? The concept of dbu doesn´t  really apply to the z-axis. If it's in µm, it definitely should be `float` not `int`

Reply:
Makes sense. But I would lean towards, don't do that here. If the dbu is bigger, it could still mean you have someayers that are very thin and can't be expressed in dbu

Reply:
ah ok